### PR TITLE
[#238] Add more metadata details in AE pipeline

### LINKED ISSUE #238: Add more metedata details in AutoEncoder pipeline
Add time measurements for epoch durations, encoding durations, evaluation durations, and all other metrics (like f1 score, accruacy etc.) to metadata.

### PR DESCRIPTION
**Pull request by Michal-Kolomanski**
closes #238

### CODE REVIEW

File: recommender/engines/autoencoders/training/data_preparation_step.py [hunk 1]
```diff
@@ -297,7 +297,7 @@ def validate_split(datasets):
             raise InvalidDatasetSplit()
 
 
-def create_details(data):
+def create_details(data: Dict) -> Dict[Dict, Dict]:
```
Comment (wujuu) [inline recommender/engines/autoencoders/training/data_preparation_step.py:300]: How can Dict be a dict key? It's not hashable, since it's mutable

File: recommender/engines/autoencoders/training/model_training_step.py [hunk 2]
```diff
@@ -104,16 +105,24 @@ def autoencoder_loss_function(reconstructions, features):
...
-
+    model = model.to(device)
     model.eval()
     with torch.no_grad():
         for batch in dataloader:
             features = batch[0].float().to(device)
-            preds = model(features).to(device)
-            loss = loss_function(features, preds)
-        return loss.item()
+            reconstructions = model(features).to(device)
```
Comment (wujuu) [inline recommender/engines/autoencoders/training/model_training_step.py:121]: Can't we use the new `get_reconstructions` function here?

Comment (wujuu) [reply recommender/engines/autoencoders/training/model_training_step.py:121]: In eval, we could even skip splitting the test dataset to batches and just evaluate the whole dataset but that may require mroe changes elsewhere

File: recommender/engines/autoencoders/training/model_validation_step.py [hunk 3]
```diff
@@ -23,9 +21,10 @@ def check_performance(metrics, max_loss_score: int) -> None:
         metrics: data from the AEModelEvaluationStep,
         max_loss_score: threshold used for validation.
     """
-
     for collection_name, metric in metrics.items():
-        for split, loss in metric.items():
+        for split, data in metric.items():
+            loss = data[LOSS]
+            # acc = data[ACCURACY]
```
Comment (wujuu) [inline recommender/engines/autoencoders/training/model_validation_step.py:27]: if we dont use accuracy in the model validation step we should just not include it, bascially, just delete the commented line

Comment (wujuu) [reply recommender/engines/autoencoders/training/model_validation_step.py:27]: Also, it might be a good idea to include it during model validation, but's im not sure its worht it now since the new AE changes are coming

**Review by wujuu — CHANGES_REQUESTED** [review]

File: recommender/engines/autoencoders/training/model_training_step.py (hunk 2 above)
Comment (wujuu) [reply recommender/engines/autoencoders/training/model_training_step.py:121]: This reminds me, we should consider including `mypy` type checker into the githooks/CI
